lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -3509,9 +3509,8 @@ public function setVersioned($bool)
     {
         $this->isVersioned = $bool;
 
-        if ($bool) {
-            $this->requiresFetchAfterChange = true;
-        }
+        $this->requiresFetchAfterChange = $bool 
+                                          || count(array_column($this->fieldMappings, 'generated')) > 0;
```

Comment:
Does $this->fieldMappings always have the same structure?

I would transfer this condition to a properly named method that describes what's going on here

Reply:
Hi!

`fieldMappings` always has type `array<string, FieldMapping>` and FieldMapping = `array<string, mixed>`. FieldMapping type describes fields of entities and contains some optional keys, but array_column is processing this cases correctly.

Check the example code:
```php
$fieldMappings = [['fieldName'=>'col1'], ['fieldName'=>'col2', 'generated'=>0]];
var_dump(array_column($fieldMappings, 'generated')); // should equals to [0=>0]

$fieldMappings = [['fieldName'=>'col1'], ['fieldName'=>'col2']];
var_dump(array_column($fieldMappings, 'generated')); // should equals to []
```

Reply:
Ok looks good, but you still did not transfer this conditional to a properly named method :>